### PR TITLE
dialect: Dropping `scf.While.get`

### PR DESCRIPTION
**Pull request by ShaolunWang**
- Fixing a bug where in `scf.While.get` we used
```Python
return While.build(
            operands=operands, result_types=result_types, regions=[before, after])
```
instead of 
```Python
return While.build(
            operands=[operands], result_types=[result_types], regions=[before, after])
```
- Marking `scf.While.get` as deprecated
- Adding respective tests for the bug and `scf.While.__init__()`

### CODE REVIEW

File: xdsl/dialects/scf.py [hunk 1]
```diff
@@ -46,6 +58,7 @@ def verify_(self):
                     f"got {self.after_region.block.args[idx].type}"
                 )
 
+    @deprecated("Use While() instead!!")
```
Comment (superlopuh) [inline xdsl/dialects/scf.py:61]:
```suggestion
    @deprecated("Use While() instead")
```

File: xdsl/dialects/scf.py [hunk 2]
```diff
@@ -30,6 +31,17 @@ class While(IRDLOperation):
     before_region: Region = region_def()
     after_region: Region = region_def()
 
+    def __init__(
+        self,
+        operands: Sequence[SSAValue | Operation],
```
Comment (superlopuh) [inline xdsl/dialects/scf.py:36]:
```suggestion
        arguments: Sequence[SSAValue | Operation],
```

File: xdsl/dialects/scf.py [hunk 3]
```diff
@@ -30,6 +31,17 @@ class While(IRDLOperation):
...
+        result_types: Sequence[Attribute],
+        before: Region | Sequence[Operation] | Sequence[Block],
```
Comment (superlopuh) [inline xdsl/dialects/scf.py:38]:
```suggestion
        before_region: Region | Sequence[Operation] | Sequence[Block],
```

File: xdsl/dialects/scf.py [hunk 4]
```diff
@@ -30,6 +31,17 @@ class While(IRDLOperation):
...
+        after: Region | Sequence[Operation] | Sequence[Block],
```
Comment (superlopuh) [inline xdsl/dialects/scf.py:39]:
```suggestion
        after_region: Region | Sequence[Operation] | Sequence[Block],
```

**Review by superlopuh — CHANGES_REQUESTED** [review]
LGTM modulo minor changes

**Review by superlopuh — APPROVED** [review]

**Review by math-fehr — APPROVED** [review]

**Pull request merged**